File: copy_display.py
# -*- coding: utf-8 -*-
import shutil
import os

src = r'D:\OneDrive\桌面\工厂图\Display Rack Workshop'
dst = r'C:\temp-display'
os.makedirs(dst, exist_ok=True)

files = sorted(os.listdir(src))
print('Source files:', files)

for i, f in enumerate(files, 1):
    src_path = os.path.join(src, f)
    ext = os.path.splitext(f)[1].lower()
    dst_name = f'display-workshop-{i}{ext}'
    dst_path = os.path.join(dst, dst_name)
    shutil.copy2(src_path, dst_path)
    print(f'Copied: {f} -> {dst_name}')

print('\nDone. Files in temp:')
for f in sorted(os.listdir(dst)):
    size = os.path.getsize(os.path.join(dst, f))
    print(f'  {f} ({size//1024}KB)')

File: copy-display.js
const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const srcDir = 'D:\\OneDrive\\桌面\\工厂图\\Display Rack Workshop';
const tempDir = 'C:\\temp-display';

if (!fs.existsSync(tempDir)) fs.mkdirSync(tempDir, { recursive: true });

const files = fs.readdirSync(srcDir);
console.log('Source files:', files);

files.forEach(f => {
  const src = path.join(srcDir, f);
  const dst = path.join(tempDir, 'display-workshop-' + f);
  const buf = fs.readFileSync(src);
  fs.writeFileSync(dst, buf);
  console.log('Copied:', f);
});

// rename to sequential
const exts = {'.jpg':'jpg','.png':'png','.jpeg':'jpeg'};
const tempFiles = fs.readdirSync(tempDir).filter(f => f.startsWith('display-workshop-'));
tempFiles.forEach((f, i) => {
  const ext = path.extname(f);
  const newName = 'display-workshop-' + (i+1) + ext.toLowerCase();
  fs.renameSync(path.join(tempDir, f), path.join(tempDir, newName));
  console.log('Renamed:', f, '->', newName);
});

console.log('DONE. Files in temp:');
fs.readdirSync(tempDir).forEach(f => console.log(' ', f));
